mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/AbstractNioWorker.java

```diff
@@ -618,4 +643,53 @@ public void run() {
             }
         });
     }
+
+    public void register(NioChildDatagramChannel childChannel) {    // TODO no datagram specific
+        registerTask(() -> {
+            childChannels.put(childChannel.getId().intValue(), childChannel);
+        });
+    }
+
+    public void deregister(NioChildDatagramChannel childChannel) {
+        registerTask(() -> {
+            childChannels.remove(childChannel.getId().intValue());
+        });
+    }
+
+    public void messageReceived(NioChildDatagramChannel childChannel, Object message) {
+        assert childChannel.getId() >= 0;
+
+        ChannelBuffer buf = (ChannelBuffer) message;
+        ByteBuffer byteBuffer = buf.toByteBuffer();
+        atomicBuffer.wrap(byteBuffer);
+        ringBuffer.write(childChannel.getId(), atomicBuffer, 0, atomicBuffer.capacity());
```

Comment:
Probably worth commenting that this write will be skipped if the buffer is full, effectively dropping the UDP packet.